### ISSUE TITLE
IDE0040 false positive for Module constructor

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#44076]
**Version Used**: 16.6 Preview 5

**Steps to Reproduce**:

```vb
Friend Module Example
  Sub New()
  End Sub
End Module
```

**Expected Behavior**:

IDE0040 is not reported.

**Actual Behavior**:

IDE0040 is reported. The code fix adds the `Private` modifier, which produces:

> BC30480: Shared 'Sub New' cannot be declared 'Private'.
